appmonitoring/ts/src/CertificateManager.ts

```diff
@@ -155,60 +155,96 @@ export class CertificateManager {
+            await logger.SendEvent("MutatingWebhookConfigurationFetchFailure", operationId, null, null, null, true, error);
             throw error;
         }
+
+        try {
+            if (!mutatingWebhookObject
+                || !mutatingWebhookObject.webhooks
+                || mutatingWebhookObject.webhooks.length !== 1 || !mutatingWebhookObject.webhooks[0].clientConfig) {
+                throw new Error("MutatingWebhookConfiguration not found or is malformed!");
+            }
+            caBundle = Buffer.from(mutatingWebhookObject.webhooks[0].clientConfig.caBundle, 'base64').toString('utf-8');
+        } catch (error) {
+            logger.error('Failed to decode caBundle from MutatingWebhookConfiguration!' + JSON.stringify(error), operationId, this.requestMetadata);
+            await logger.SendEvent("MutatingWebhookCABundleDecodeFailure", operationId, null, null, null, true, error);
+
+            caBundle = null; // return null to indicate failure in decoding
+            logger.info('Returning null caBundle due to decode failure.', operationId, this.requestMetadata);
+        }
+
+        return caBundle;
     }
 
-    public async PatchMutatingWebhook(operationId: string, kubeConfig: k8s.KubeConfig, certificate: WebhookCertData) {
+    public async UpdateMutatingWebhook(operationId: string, kubeConfig: k8s.KubeConfig, certificate: WebhookCertData): Promise<void> {
         try {
             const webhookApi: k8s.AdmissionregistrationV1Api = kubeConfig.makeApiClient(k8s.AdmissionregistrationV1Api);
             const mutatingWebhookObject: k8s.V1MutatingWebhookConfiguration = await webhookApi.readMutatingWebhookConfiguration({ name: MutatingWebhookConfigurationName });
-            if (!mutatingWebhookObject 
-                || !mutatingWebhookObject.webhooks 
-                || mutatingWebhookObject.webhooks.length !== 1 || !mutatingWebhookObject.webhooks[0].clientConfig)
-            {
+            if (!mutatingWebhookObject
+                || !mutatingWebhookObject.webhooks
+                || mutatingWebhookObject.webhooks.length !== 1 || !mutatingWebhookObject.webhooks[0].clientConfig) {
                 throw new Error("MutatingWebhookConfiguration not found or is malformed!");
             }
             mutatingWebhookObject.webhooks[0].clientConfig.caBundle = Buffer.from(certificate.caCert, 'utf-8').toString('base64');
-            
+
             await webhookApi.replaceMutatingWebhookConfiguration({ name: MutatingWebhookConfigurationName, body: mutatingWebhookObject });
         } catch (error) {
-            logger.error('Failed to patch MutatingWebhookConfiguration!', operationId, this.requestMetadata);
-            logger.error(JSON.stringify(error), operationId, this.requestMetadata);
+            logger.error('Failed to patch MutatingWebhookConfiguration!' + JSON.stringify(error), operationId, this.requestMetadata);
```

Comment:
`${}`

Reply:
hmm I did not realize you could do it like that